www/source/blog/2017-10-20-node-scaffolding-scripts.html.md

```diff
@@ -0,0 +1,97 @@
+---
+title: "Node Scaffolding now supports pre and post build scripts!"
+date: 2017-10-20
+author: Nell Shamrell-Harrington
+tags: Node, scaffolding
+category: community
+classes: body-article
+---
+
+We have just released a new version of the Node scaffolding that supports scripts defined in package.json!
+
+It is common for Node applications to include scripts in their package.json files like so:
+
+**package.json**
+
+```
+"scripts": {
+	"start": "react-scripts start",
+	"build": "react-scripts build",
+	"test": "react-scripts test --env=jsdom",
+	"eject": "react-scripts eject"
+}
+```
+
+The [Habitat Node Scaffolding](https://github.com/habitat-sh/core-plans/tree/master/scaffolding-node) now supports these scripts out of the box.
+
+Let's take it for a test run:
+
+One of my favorite sample React apps is the [Pokedex](https://alik0211.ru/pokedex/) by Ali Gasymov.
+
+Let's habiterize this application and deploy it in a Docker container!
+
+First, clone the Github repo:
+
+```console
+$ git clone git@github.com:alik0211/pokedex.git
+$ cd pokedex
+```
+
+Now install the dependencies:
+
+```console
+$ npm install
+```
+
+Check out the package.json - notice the scripts section? We will be able to run these to start, build, etc. the application through the Node Scaffolding.
+
+**package.json***
+
+```
+{
+	"name": "pokedex",
+		"version": "1.1.5",
+		"private": true,
+		"dependencies": {
+			"react": "^16.0.0",
+			"react-dom": "^16.0.0",
+			"react-redux": "^5.0.6",
+			"redux": "^3.7.2",
+			"redux-thunk": "^2.2.0"
+		},
+		"devDependencies": {
+			"react-scripts": "1.0.10",
+			"redux-logger": "^3.0.6"
+		},
+		"scripts": {
+			"start": "react-scripts start",
+			"build": "react-scripts build",
+			"test": "react-scripts test --env=jsdom",
+			"eject": "react-scripts eject"
+		}
+}
+```
+
+
+Let's Habitize this app:
+
+```
+$ hab plan init -s node
+$ hab studio enter
+$ (studio) build
```

Comment:
I know the answer of course, but a reader new to Habitat might wonder whether you mean to type `(studio)` here, since it's after the `$` and in the same output block.

Reply:
This worked (yay!), but it might be worth pointing out what I'm looking for to see that these scripts actually did something. I can't quite tell what I should see after the build completes. Just that the app is running? (I thought we already supported `npm start`, but I might be wrong about that.)